src/mpid/ch4/src/ch4_impl.h

```diff
@@ -429,12 +429,6 @@ static inline int MPIDIU_valid_group_rank(MPIR_Comm * comm, int rank, MPIR_Group
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDIU_VALID_GROUP_RANK);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIU_VALID_GROUP_RANK);
 
-    if (unlikely(rank == MPI_PROC_NULL)) {
-        /* Treat PROC_NULL as always valid */
-        ret = 1;
-        goto fn_exit;
-    }
-
```

Comment:
What do you think about this comment ?

Reply:
I think I initially restored this check after your comment. But later since we have included non-RMA functions in this PR, we can safely remove this check now, right?

Reply:
It is a utility function. For now it is used only in the RMA checking. But maybe in future it will be used in a function which has to handle PROC_NULL. How about we add a comment here saying that it assumes the calling routine never checks `PROC_NULL` ?

Reply:
I see. I have added the comment.